### PR TITLE
Support multiple belongsTo for next branch

### PR DESCRIPTION
**Pull request by MarlonJD**
*Same Issue #2277 and #1814 on the amplify_api 1.0.0-next.1,

*Description of changes:*
It's #2087 for next branch. Thanks for @fjnoyp

### CODE REVIEW

**Comment by ragingsquirrel3** [comment]
@MarlonJD thanks for opening this. Request: could you please cherry pick the commit from main rather than making new commits with same changes? You should just be able to run `git cherry-pick 7c14f79fb247ec8725f8fdf7a6a86ce5ea47494e` to get the changes from https://github.com/aws-amplify/amplify-flutter/commit/7c14f79fb247ec8725f8fdf7a6a86ce5ea47494e . That way, the same commit will be on both branches which helps to organize the git history.

**Comment by ragingsquirrel3** [comment]
@MarlonJD yes would either fix conflicts while doing cherry-pick (and callout in PR diff where the conflicts were), or alternatively add 1 extra commit after the cherry-picked commit whose commit message makes it clear that it is fixing code from previous commit. If you add an extra commit, would suggest to keep amending that commit for all the other changes in this PR so that this PR would have max 2 commits. Like I said, the end goal is that we have the original commit from main on next. thanks again!

**Comment by MarlonJD** [comment]
Okay thanks for your detailed answer. So I'll  do cherry pick and merge commit then create pr again

**Comment by MarlonJD** [comment]
@ragingsquirrel3 I did what you described, merged 2 confict. Can you check ?

**Comment by ragingsquirrel3** [comment]
Thanks, will review!

File: packages/api/amplify_api/test/graphql_helpers_mtm_test.dart [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+  MockAmplifyAPI({
+    ModelProviderInterface? modelProvider,
+  }) : super(modelProvider: modelProvider);
+
+  @override
+  void registerAuthProvider(APIAuthProvider authProvider) {}
+
+  @override
+  Future<void> addPlugin() async {}
+}
```
Comment (ragingsquirrel3) [inline packages/api/amplify_api/test/graphql_helpers_mtm_test.dart:38]:
```suggestion
class MockAmplifyAPI extends AmplifyAPIDart {
  MockAmplifyAPI({
    super.modelProvider,
  });
}
```

File: packages/api/amplify_api/test/graphql_helpers_mtm_test.dart [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                }                
+              }
+            ]
+          }
+        }''';
+
+      GraphQLResponse<PaginatedResult<FirstMtmRelation>> response =
+          GraphQLResponseDecoder.instance
+              .decode<PaginatedResult<FirstMtmRelation>>(
+                  request: req, data: data, errors: errors);
```
Comment (ragingsquirrel3) [inline packages/api/amplify_api/test/graphql_helpers_mtm_test.dart:180]:
```suggestion
      const data = {
        'listFirstMtmRelations': {
          'items': [
            {
              'id': 'id-first-mtm-relation',
              'manyToManyPrimary': {
                'id': 'id-mtm-primary',
                'name': 'mtm primary'
              },
              'manyToManySecondary': {
                'id': 'id-mtm-secondary',
                'name': 'mtm secondary'
              }
            }
          ]
        }
      };

      final response = GraphQLResponseDecoder.instance
          .decode<PaginatedResult<FirstMtmRelation>>(
        request: req,
        response: {
          'data': data,
          'errors': null,
        },
      );
```

**Review by ragingsquirrel3 — CHANGES_REQUESTED** [review]
Thanks again for doing this! I just left a few suggestions regarding updates needed for next branch, might have been related to conflicts. You can create a new commit with title "chore(api): update conflicts from #2087" with these changes. There will also be some formatting changes needed as lint requirements are stricter on this branch. To check those, you can run `cd packages/api/amplify_api && flutter analyze`. I will also enable the CI checks for you.

**Review by dnys1 — APPROVED** [review]

**Comment by MarlonJD** [comment]
I'm trying to fix conficts and prepairing changes.

**Comment by MarlonJD** [comment]
Rebase changes for only for 2 commit

**Comment by ragingsquirrel3** [comment]
@MarlonJD good question. The contributing tips are a little different on next. You have to bootstrap the repo with `aft bs` instead of `melos bootstrap`:
* from root of repo, install `aft` w command `dart pub global activate -spath packages/aft`
* bootstrap by running `aft bs`

That will build the worker JS file locally.
More info https://github.com/aws-amplify/amplify-flutter/blob/next/CONTRIBUTING.md#amplify-flutter-repo-tool-aft

**Comment by ragingsquirrel3** [comment]
thanks @MarlonJD, tests passed locally for me. This is looking pretty good on my end. I'm just checking a few things then will approve/merge or let you know if other changes requested.

**Review by ragingsquirrel3 — APPROVED** [review]
thanks again @MarlonJD . Also thanks for bearing w me on cherry picking the commit and correcting for updated lint rules.

**Comment by MarlonJD** [comment]
@ragingsquirrel3 thank you for your support and patience. Happy to see it merged.

**Pull request merged**